src/POData/UriProcessor/RequestDescription.php

```diff
@@ -311,9 +311,10 @@ private function readData($dataType)
                 SerializerBuilder::create()
                     ->addMetadataDir($ymlDir)
                     ->build();
-            $this->data = $serialize->deserialize($string, 'POData\ObjectModel\ODataEntry', 'xml');
-            assert($this->data instanceof ODataEntry);
```

Comment:
Can you resurrect this assert as ($this->date instanceof $objectType), immediately after defining $objectType - line 315 or so?